iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/IcebergTableUtil.java

```diff
@@ -473,17 +473,22 @@ public static Expression generateExpressionFromPartitionSpec(Table table, Map<St
     return finalExp;
   }
 
-  public static List<FieldSchema> getPartitionKeys(Table table, int specId) {
-    Schema schema = table.specs().get(specId).schema();
-    List<FieldSchema> hiveSchema = HiveSchemaUtil.convert(schema);
-    Map<String, String> colNameToColType = hiveSchema.stream()
-        .collect(Collectors.toMap(FieldSchema::getName, FieldSchema::getType));
-    return table.specs().get(specId).fields().stream().map(partField ->
-        new FieldSchema(schema.findColumnName(partField.sourceId()),
-            colNameToColType.get(schema.findColumnName(partField.sourceId())),
-            String.format("Transform: %s", partField.transform().toString()))).collect(Collectors.toList());
+  public static List<FieldSchema> getPartitionKeys(Table table, boolean latestSpecOnly) {
```

Comment:
that is definitely less readable than the prev version

Reply:
Refactored to the previous version.